lib/IRGen/DebugTypeInfo.h

```diff
@@ -78,25 +78,29 @@ class DebugTypeInfo {
 
   void unwrapLValueOrInOutType() {
     Type = Type->getLValueOrInOutObjectType().getPointer();
-    }
+  }
 
-    // Determine whether this type is an Archetype itself.
-    bool isArchetype() const {
-      return Type->getLValueOrInOutObjectType()->is<ArchetypeType>();
-    }
+  // Determine whether this type is an Archetype itself.
+  bool isArchetype() const {
+    return Type->getLValueOrInOutObjectType()->is<ArchetypeType>();
+  }
 
-    /// LValues, inout args, and Archetypes are implicitly indirect by
-    /// virtue of their DWARF type.
-    bool isImplicitlyIndirect() const {
-      return Type->isLValueType() || isArchetype() ||
-             (Type->getKind() == TypeKind::InOut);
-    }
+  /// LValues, inout args, and Archetypes are implicitly indirect by
+  /// virtue of their DWARF type.
+  //
+  // FIXME: Should this check if the lowered SILType is address only
```

Comment:
@adrian-prantl What do you think? Should we be checking the SIL type lowering instead?

Also class-bound archetypes are not passed indirectly, so it's wrong in that case too.